### PR TITLE
add /api/users GET routes

### PR DESCRIPTION
**Pull request by nathanperkins**
- added a router for `/api`, and added it to the main router, below the api routes that are already in place.
- added a router for `/api/users/`
- GET `/api/users` will present a list of users, with password removed.
-GET `/api/users/current` will present the current user.
- GET `/api/users/:userId` will present a specific user based on their id, but only the id matches the current user. This can be used later to allow admins to view a user.

Strech idea:
- Reduce the duplicated code between the current and :userId routes.

### CODE REVIEW

File: routes/api/users.js [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+            res.json({
+                user: user,
+                courses: courses,
+                pageTitle: `${user.firstName} ${user.lastName}`,
+            });
+        }).catch(function (err) {
+            console.log(`User request error: ${err}`);
+            res.json({
+                pageTitle: pageTitle,
+                error: errmsg,
```
Comment (toucan-stan) [inline routes/api/users.js]: line 54: should this be err.errmsg?

**Review by toucan-stan — COMMENTED** [review]
General comment: is pageTitle necessary to return from the API, given that we're using angular? If we're just returning JSON, seems that the actual page title will be handled by the front end.

**Comment by nathanperkins** [comment]
I think they should be removed from the API, asking @sarahmaas.

**Comment by nathanperkins** [comment]
ill remove the page titles

**Comment by nathanperkins** [comment]
I've removed the pageTitles and added a logged in user validation for the `/api/courses` route

**Pull request merged**